client/dashboard/me/billing-purchases/purchase-settings/index.tsx

```diff
@@ -0,0 +1,1209 @@
+	const canBeRenewed =
+		purchase.can_explicit_renew && String( user.ID ) === String( purchase.user_id );
+	const { recordTracksEvent } = useAnalytics();
+	if ( ! canBeRenewed ) {
+		return null;
+	}
+
+	return (
+		<ActionList.ActionItem
+			title={ __( 'Renew now' ) }
+			description={ __( 'Renew your subscription manually.' ) }
+			actions={
+				<Button
+					variant="secondary"
+					size="compact"
+					onClick={ () => {
+						recordTracksEvent( 'calypso_purchases_renew_now_click', {
+							product_slug: purchase.product_slug,
+						} );
+						renewPurchase( purchase );
+					} }
+				>
+					{ __( 'Renew' ) }
+				</Button>
+			}
+		/>
+	);
+}
+
+function JetpackCRMDownloadsButton( { purchase }: { purchase: Purchase } ) {
+	const { recordTracksEvent } = useAnalytics();
+
+	// Only show for Jetpack CRM Products
+	if ( ! isJetpackCrmProduct( purchase.product_slug ) ) {
+		return null;
+	}
+
+	// We'll pass the purchase ID in the URL, and the CRM Downloads component will fetch the actual license key
+	const path = `/purchases/crm-downloads/${ purchase.ID }`;
+
+	return (
+		<ActionList.ActionItem
+			title={ __( 'CRM Downloads' ) }
+			actions={
+				<Button
+					variant="secondary"
+					size="compact"
+					onClick={ () => {
+						recordTracksEvent( 'calypso_purchases_crm_downloads_click', {
+							product_slug: purchase.product_slug,
+						} );
+						window.location.href = path;
+					} }
+				>
+					{ __( 'CRM Downloads' ) }
+				</Button>
+			}
+		/>
+	);
+}
+
+function ReinstallButton( { purchase }: { purchase: Purchase } ) {
+	const {
+		mutate: reinstallPlugins,
+		error,
+		isPending: isMutationPending,
+	} = useMutation( reinstallMarketplacePluginsQuery( parseInt( purchase.blog_id ) ) );
+	if ( ! isMarketplacePlugin( purchase ) ) {
+		return null;
+	}
+	if ( isMarketplaceTemporarySitePurchase( purchase ) ) {
+		return null;
+	}
+
+	return (
+		<ActionList.ActionItem
+			title={ __( 'Reinstall plugins' ) }
+			actions={
+				<>
+					<Button
+						variant="secondary"
+						size="compact"
+						disabled={ isMutationPending }
+						onClick={ () => {
+							reinstallPlugins();
+						} }
+					>
+						{ __( 'Reinstall plugins' ) }
+					</Button>
+					{
+						// FIXME: there's probably a better place for this error message
+						error && (
+							<Notice status="error" isDismissible={ false }>
+								{ error.message }
+							</Notice>
+						)
+					}
+				</>
+			}
+		/>
+	);
+}
+
+function PurchaseSettingsActions( { purchase }: { purchase: Purchase } ) {
+	return (
+		<VStack spacing={ 4 }>
+			<ActionList>
+				<ReinstallButton purchase={ purchase } />
+				<JetpackCRMDownloadsButton purchase={ purchase } />
+				<UpgradeActionButton purchase={ purchase } />
+				<ReSubscribeActionButton purchase={ purchase } />
+				<RenewActionButton purchase={ purchase } />
+				<CancelOrRemoveActionButton purchase={ purchase } />
+			</ActionList>
+		</VStack>
+	);
+}
+
+function PurchaseSettingsCard( {
```

Comment:
Early comment: can we just reuse the existing [OverviewCard](https://github.com/Automattic/wp-calypso/blob/c3875a1f60b35ad67767fc2f50926f8ddb9d4d3f/client/dashboard/sites/overview-card/index.tsx)? Based on the design, seems like they should be the same thing as the one in the Site Overview.

I see that the component is inside `sites/` -- maybe we should extract that to a higher-level `components/` directory. cc @Automattic/lego

Reply:
I'll try to do it when I have spare time.